drivers/sensor/ltc4162/ltc4162.c

```diff
@@ -0,0 +1,430 @@
+		break;
+
+	case SENSOR_CHAN_CHARGER_INPUT_CURRENT:
+		read_addr = LTC4162_IIN;
+		ret = i2c_burst_read_dt(&cfg->bus, read_addr, &buf[0],
+					sizeof(buf));
+		if (ret < 0) {
+			LOG_ERR("Failed to read input current");
+			goto error;
+		}
+
+		regval = (buf[1] << 8 | buf[0]);
+		temp_regval = (regval * 1.466) / (cfg->rsnsi);
+		data->in_current = (uint16_t)temp_regval;
+		break;
+
+	case SENSOR_CHAN_CHARGER_CONSTANT_CHARGE_CURRENT:
+		read_addr = LTC4162_ICHARGE_DAC;
+		ret = i2c_burst_read_dt(&cfg->bus, read_addr, &buf[0],
+					sizeof(buf));
+		if (ret < 0) {
+			LOG_ERR("Failed to read constant current");
+			goto error;
+		}
+
+		regval = (buf[1] << 8 | buf[0]);
+		regval &= BIT(6) - 1;
+		++regval;
+		temp_regval = ((float)regval / cfg->rsnsb);
+		data->const_current = (temp_regval * 10000);
+		break;
+
+	case SENSOR_CHAN_CHARGER_CONSTANT_CHARGE_CURRENT_MAX:
+		read_addr = LTC4162_CHARGE_CURRENT_SETTING;
+		ret = i2c_burst_read_dt(&cfg->bus, read_addr, &buf[0],
+					sizeof(buf));
+		if (ret < 0) {
+			LOG_ERR("Failed to read constant current max");
+			goto error;
+		}
+
+		regval = (buf[1] << 8 | buf[0]);
+		regval &= BIT(6) - 1;
+		++regval;
+		temp_regval = ((float)regval / cfg->rsnsb);
+		data->const_current_max = (temp_regval * 10000);
+		break;
+
+	case SENSOR_CHAN_CHARGER_CONSTANT_CHARGE_VOLTAGE:
+		read_addr = LTC4162_VCHARGE_DAC;
+		ret = i2c_burst_read_dt(&cfg->bus, read_addr, &buf[0],
+					sizeof(buf));
+		if (ret < 0) {
+			LOG_ERR("Failed to read constant charge voltage");
+			goto error;
+		}
+
+		regval = (buf[1] << 8 | buf[0]);
+		regval &= BIT(6) - 1;
+		temp_regval = ((float)regval * CHRG_VOLTAGE_OFFSET);
+		temp_regval = cfg->cell_count * (temp_regval + 6);
+		data->const_voltage = (temp_regval * 1000);
+		break;
+
+	case SENSOR_CHAN_CHARGER_CONSTANT_CHARGE_VOLTAGE_MAX:
+		read_addr = LTC4162_VCHARGE_SETTING;
+		ret = i2c_burst_read_dt(&cfg->bus, read_addr, &buf[0],
+					sizeof(buf));
+		if (ret < 0) {
+			LOG_ERR("Failed to read constant charge current");
+			goto error;
+		}
+
+		regval = (buf[1] << 8 | buf[0]);
+		regval &= BIT(6) - 1;
+		temp_regval = ((float)regval * CHRG_VOLTAGE_OFFSET);
+		temp_regval = cfg->cell_count * (temp_regval + 6);
+		data->const_voltage_max = (temp_regval * 1000);
+		break;
+
+	case SENSOR_CHAN_CHARGER_INPUT_CURRENT_LIMIT:
+		read_addr = LTC4162_IIN_LIMIT_DAC;
+		ret = i2c_burst_read_dt(&cfg->bus, read_addr, &buf[0],
+					sizeof(buf));
+		if (ret < 0) {
+			LOG_ERR("Failed to read constant current");
+			goto error;
+		}
+
+		regval = (buf[1] << 8 | buf[0]);
+		regval &= BIT(6) - 1;
+		++regval;
+		temp_regval = ((float)regval / 2);
+		temp_regval = (temp_regval / cfg->rsnsi);
+		data->in_current_limit = (temp_regval * 1000);
+		break;
+
+	case SENSOR_CHAN_CHARGER_TEMPERATURE:
+		read_addr = LTC4162_DIE_TEMPERATURE;
+		ret = i2c_burst_read_dt(&cfg->bus, read_addr, &buf[0],
+					sizeof(buf));
+		if (ret < 0) {
+			LOG_ERR("Failed to read constant current");
+			goto error;
```

Comment:
`goto` should be used if resources need to be released, e.g. when a mutex needs to be unlocked before returning. In this case a simple return is enough.